builtins.go

```diff
@@ -1161,6 +1163,106 @@ func builtinParseJSON(i *interpreter, trace traceElement, str value) (value, err
 	return jsonToValue(i, trace, parsedJSON)
 }
 
+// We have a very similar logic here /interpreter.go@v0.16.0#L695 and here: /interpreter.go@v0.16.0#L627
+// These should ideally be unified
+// For backwards compatibility reasons, we are manually marshalling to json so we can control formatting
+// In the future, it might be apt to use a library [pretty-printing] function
+func builtinManifestJSONEx(i *interpreter, trace traceElement, obj, indent value) (value, error) {
+	vindent, err := i.getString(indent, trace)
+	if err != nil {
+		return nil, err
+	}
+
+	sindent := vindent.getGoString()
+
+	var path []string
+
+	var aux func(ov value, path []string, cindent string) (string, error)
+	aux = func(ov value, path []string, cindent string) (string, error) {
+		if ov == nil {
+			fmt.Println("value is nil")
+			return "null", nil
+		}
+
+		switch v := ov.(type) {
+		case *valueNull:
+			return "null", nil
+		case valueString:
+			b, err := json.Marshal(v.getGoString())
+			if err != nil {
+				return "", i.Error(fmt.Sprintf("failed to marshal valueString to JSON: %v", err.Error()), trace)
+			}
+			return string(b), nil
+		case *valueNumber:
+			return strconv.FormatFloat(v.value, 'f', -1, 64), nil
+		case *valueBoolean:
+			return fmt.Sprintf("%t", v.value), nil
+		case *valueFunction:
+			return "", i.Error(fmt.Sprintf("tried to manifest function at %s", path), trace)
+		case *valueArray:
+			newIndent := cindent + sindent
+			lines := []string{"[\n"}
+
+			var arrayLines []string
+			for aI, cThunk := range v.elements {
+				cTv, err := cThunk.getValue(i, trace)
+				if err != nil {
+					return "", err
+				}
+
+				newPath := append(path, string(aI))
```

Comment:
string(aI) doesn't do what you think. See: https://play.golang.org/p/UkFqprKKRm4. You probably want strconv.FormatInt(aI).

Reply:
fixed. Sounds like maybe I should write some tests!